File: .sdk-repos/version-8.9/orchestration-cluster-api-python/examples/tenant.py
# Compilable usage examples for tenant management operations.
# These examples are type-checked during build to guard against API regressions.
from __future__ import annotations

from camunda_orchestration_sdk import (
    CamundaClient,
    MappingRuleSearchQueryRequest,
    RoleSearchQueryRequest,
    TenantCreateRequest,
    TenantGroupSearchQueryRequest,
    TenantId,
    TenantSearchQueryRequest,
    TenantUpdateRequest,
    Unset,
    Username,
)


# region CreateTenant
def create_tenant_example(tenant_id: TenantId) -> None:
    client = CamundaClient()

    result = client.create_tenant(
        data=TenantCreateRequest(
            tenant_id=tenant_id,
            name="Acme Corporation",
        ),
    )

    print(f"Tenant: {result.tenant_id}")
# endregion CreateTenant


# region GetTenant
def get_tenant_example(tenant_id: TenantId) -> None:
    client = CamundaClient()

    result = client.get_tenant(tenant_id=tenant_id)

    print(f"Tenant: {result.name}")
# endregion GetTenant


# region SearchTenants
def search_tenants_example() -> None:
    client = CamundaClient()

    result = client.search_tenants(
        data=TenantSearchQueryRequest(),
    )

    if not isinstance(result.items, Unset):
        for tenant in result.items:
            print(f"Tenant: {tenant.name}")
# endregion SearchTenants


# region UpdateTenant
def update_tenant_example(tenant_id: TenantId) -> None:
    client = CamundaClient()

    client.update_tenant(
        tenant_id=tenant_id,
        data=TenantUpdateRequest(name="Acme Corp International"),
    )
# endregion UpdateTenant


# region DeleteTenant
def delete_tenant_example(tenant_id: TenantId) -> None:
    client = CamundaClient()

    client.delete_tenant(tenant_id=tenant_id)
# endregion DeleteTenant


# region AssignUserToTenant
def assign_user_to_tenant_example(tenant_id: TenantId, username: Username) -> None:
    client = CamundaClient()

    client.assign_user_to_tenant(
        tenant_id=tenant_id,
        username=username,
    )
# endregion AssignUserToTenant


# region UnassignUserFromTenant
def unassign_user_from_tenant_example(tenant_id: TenantId, username: Username) -> None:
    client = CamundaClient()

    client.unassign_user_from_tenant(
        tenant_id=tenant_id,
        username=username,
    )
# endregion UnassignUserFromTenant


# region AssignGroupToTenant
def assign_group_to_tenant_example(tenant_id: TenantId) -> None:
    client = CamundaClient()

    client.assign_group_to_tenant(
        tenant_id=tenant_id,
        group_id="engineering",
    )
# endregion AssignGroupToTenant


# region UnassignGroupFromTenant
def unassign_group_from_tenant_example(tenant_id: TenantId) -> None:
    client = CamundaClient()

    client.unassign_group_from_tenant(
        tenant_id=tenant_id,
        group_id="engineering",
    )
# endregion UnassignGroupFromTenant


# region AssignRoleToTenant
def assign_role_to_tenant_example(tenant_id: TenantId) -> None:
    client = CamundaClient()

    client.assign_role_to_tenant(
        tenant_id=tenant_id,
        role_id="developer",
    )
# endregion AssignRoleToTenant


# region UnassignRoleFromTenant
def unassign_role_from_tenant_example(tenant_id: TenantId) -> None:
    client = CamundaClient()

    client.unassign_role_from_tenant(
        tenant_id=tenant_id,
        role_id="developer",
    )
# endregion UnassignRoleFromTenant


# region AssignClientToTenant
def assign_client_to_tenant_example(tenant_id: TenantId) -> None:
    client = CamundaClient()

    client.assign_client_to_tenant(
        tenant_id=tenant_id,
        client_id="my-service-account",
    )
# endregion AssignClientToTenant


# region UnassignClientFromTenant
def unassign_client_from_tenant_example(tenant_id: TenantId) -> None:
    client = CamundaClient()

    client.unassign_client_from_tenant(
        tenant_id=tenant_id,
        client_id="my-service-account",
    )
# endregion UnassignClientFromTenant


# region AssignMappingRuleToTenant
def assign_mapping_rule_to_tenant_example(tenant_id: TenantId) -> None:
    client = CamundaClient()

    client.assign_mapping_rule_to_tenant(
        tenant_id=tenant_id,
        mapping_rule_id="rule-123",
    )
# endregion AssignMappingRuleToTenant


# region UnassignMappingRuleFromTenant
def unassign_mapping_rule_from_tenant_example(tenant_id: TenantId) -> None:
    client = CamundaClient()

    client.unassign_mapping_rule_from_tenant(
        tenant_id=tenant_id,
        mapping_rule_id="rule-123",
    )
# endregion UnassignMappingRuleFromTenant


# region SearchUsersForTenant
def search_users_for_tenant_example(tenant_id: TenantId) -> None:
    client = CamundaClient()

    result = client.search_users_for_tenant(
        tenant_id=tenant_id,
    )

    if not isinstance(result.items, Unset):
        for user in result.items:
            print(f"User: {user.username}")
# endregion SearchUsersForTenant


# region SearchClientsForTenant
def search_clients_for_tenant_example(tenant_id: TenantId) -> None:
    client = CamundaClient()

    result = client.search_clients_for_tenant(
        tenant_id=tenant_id,
    )

    if not isinstance(result.items, Unset):
        for c in result.items:
            print(f"Client: {c.client_id}")
# endregion SearchClientsForTenant


# region SearchGroupIdsForTenant
def search_group_ids_for_tenant_example(tenant_id: TenantId) -> None:
    client = CamundaClient()

    result = client.search_group_ids_for_tenant(
        tenant_id=tenant_id,
        data=TenantGroupSearchQueryRequest(),
    )

    if not isinstance(result.items, Unset):
        for group in result.items:
            print(f"Group: {group.group_id}")
# endregion SearchGroupIdsForTenant


# region SearchRolesForTenant
def search_roles_for_tenant_example(tenant_id: TenantId) -> None:
    client = CamundaClient()

    result = client.search_roles_for_tenant(
        tenant_id=tenant_id,
        data=RoleSearchQueryRequest(),
    )

    if not isinstance(result.items, Unset):
        for role in result.items:
            print(f"Role: {role.name}")
# endregion SearchRolesForTenant


# region SearchMappingRulesForTenant
def search_mapping_rules_for_tenant_example(tenant_id: TenantId) -> None:
    client = CamundaClient()

    result = client.search_mapping_rules_for_tenant(
        tenant_id=tenant_id,
        data=MappingRuleSearchQueryRequest(),
    )

    if not isinstance(result.items, Unset):
        for rule in result.items:
            print(f"Mapping rule: {rule.mapping_rule_id}")
# endregion SearchMappingRulesForTenant

File: .sdk-repos/version-8.9/orchestration-cluster-api-csharp/examples/Tenant.cs
// Compilable usage examples for tenant management operations.
// These examples are type-checked during build to guard against API regressions.
using Camunda.Orchestration.Sdk;

public static class TenantExamples
{
    #region CreateTenant
    // <CreateTenant>
    public static async Task CreateTenantExample()
    {
        using var client = CamundaClient.Create();

        var result = await client.CreateTenantAsync(new TenantCreateRequest
        {
            TenantId = "acme-corp",
            Name = "Acme Corporation",
        });

        Console.WriteLine($"Tenant key: {result.TenantId}");
    }
    // </CreateTenant>
    #endregion CreateTenant

    #region GetTenant

    // <GetTenant>
    public static async Task GetTenantExample(TenantId tenantId)
    {
        using var client = CamundaClient.Create();

        var result = await client.GetTenantAsync(tenantId);
        Console.WriteLine($"Tenant: {result.Name}");
    }
    // </GetTenant>
    #endregion GetTenant

    #region SearchTenants

    // <SearchTenants>
    public static async Task SearchTenantsExample()
    {
        using var client = CamundaClient.Create();

        var result = await client.SearchTenantsAsync(new TenantSearchQueryRequest());

        foreach (var tenant in result.Items)
        {
            Console.WriteLine($"Tenant: {tenant.Name}");
        }
    }
    // </SearchTenants>
    #endregion SearchTenants

    #region UpdateTenant

    // <UpdateTenant>
    public static async Task UpdateTenantExample(TenantId tenantId)
    {
        using var client = CamundaClient.Create();

        await client.UpdateTenantAsync(
            tenantId,
            new TenantUpdateRequest
            {
                Name = "Acme Corp International",
            });
    }
    // </UpdateTenant>
    #endregion UpdateTenant

    #region DeleteTenant

    // <DeleteTenant>
    public static async Task DeleteTenantExample(TenantId tenantId)
    {
        using var client = CamundaClient.Create();

        await client.DeleteTenantAsync(tenantId);
    }
    // </DeleteTenant>
    #endregion DeleteTenant

    #region AssignUserToTenant

    // <AssignUserToTenant>
    public static async Task AssignUserToTenantExample(TenantId tenantId, Username username)
    {
        using var client = CamundaClient.Create();

        await client.AssignUserToTenantAsync(
            tenantId,
            username);
    }
    // </AssignUserToTenant>
    #endregion AssignUserToTenant

    #region UnassignUserFromTenant

    // <UnassignUserFromTenant>
    public static async Task UnassignUserFromTenantExample(TenantId tenantId, Username username)
    {
        using var client = CamundaClient.Create();

        await client.UnassignUserFromTenantAsync(
            tenantId,
            username);
    }
    // </UnassignUserFromTenant>
    #endregion UnassignUserFromTenant

    #region AssignGroupToTenant

    // <AssignGroupToTenant>
    public static async Task AssignGroupToTenantExample(TenantId tenantId)
    {
        using var client = CamundaClient.Create();

        await client.AssignGroupToTenantAsync(
            tenantId,
            "engineering");
    }
    // </AssignGroupToTenant>
    #endregion AssignGroupToTenant

    #region UnassignGroupFromTenant

    // <UnassignGroupFromTenant>
    public static async Task UnassignGroupFromTenantExample(TenantId tenantId)
    {
        using var client = CamundaClient.Create();

        await client.UnassignGroupFromTenantAsync(
            tenantId,
            "engineering");
    }
    // </UnassignGroupFromTenant>
    #endregion UnassignGroupFromTenant

    #region AssignRoleToTenant

    // <AssignRoleToTenant>
    public static async Task AssignRoleToTenantExample(TenantId tenantId)
    {
        using var client = CamundaClient.Create();

        await client.AssignRoleToTenantAsync(
            tenantId,
            "developer");
    }
    // </AssignRoleToTenant>
    #endregion AssignRoleToTenant

    #region UnassignRoleFromTenant

    // <UnassignRoleFromTenant>
    public static async Task UnassignRoleFromTenantExample(TenantId tenantId)
    {
        using var client = CamundaClient.Create();

        await client.UnassignRoleFromTenantAsync(
            tenantId,
            "developer");
    }
    // </UnassignRoleFromTenant>
    #endregion UnassignRoleFromTenant

    #region AssignClientToTenant

    // <AssignClientToTenant>
    public static async Task AssignClientToTenantExample(TenantId tenantId)
    {
        using var client = CamundaClient.Create();

        await client.AssignClientToTenantAsync(
            tenantId,
            "my-service-account");
    }
    // </AssignClientToTenant>
    #endregion AssignClientToTenant

    #region UnassignClientFromTenant

    // <UnassignClientFromTenant>
    public static async Task UnassignClientFromTenantExample(TenantId tenantId)
    {
        using var client = CamundaClient.Create();

        await client.UnassignClientFromTenantAsync(
            tenantId,
            "my-service-account");
    }
    // </UnassignClientFromTenant>
    #endregion UnassignClientFromTenant

    #region AssignMappingRuleToTenant

    // <AssignMappingRuleToTenant>
    public static async Task AssignMappingRuleToTenantExample(TenantId tenantId)
    {
        using var client = CamundaClient.Create();

        await client.AssignMappingRuleToTenantAsync(
            tenantId,
            "rule-123");
    }
    // </AssignMappingRuleToTenant>
    #endregion AssignMappingRuleToTenant

    #region UnassignMappingRuleFromTenant

    // <UnassignMappingRuleFromTenant>
    public static async Task UnassignMappingRuleFromTenantExample(TenantId tenantId)
    {
        using var client = CamundaClient.Create();

        await client.UnassignMappingRuleFromTenantAsync(
            tenantId,
            "rule-123");
    }
    // </UnassignMappingRuleFromTenant>
    #endregion UnassignMappingRuleFromTenant

    #region SearchUsersForTenant

    // <SearchUsersForTenant>
    public static async Task SearchUsersForTenantExample(TenantId tenantId)
    {
        using var client = CamundaClient.Create();

        var result = await client.SearchUsersForTenantAsync(
            tenantId,
            new SearchUsersForTenantRequest());

        foreach (var user in result.Items)
        {
            Console.WriteLine($"User: {user.Username}");
        }
    }
    // </SearchUsersForTenant>
    #endregion SearchUsersForTenant

    #region SearchClientsForTenant

    // <SearchClientsForTenant>
    public static async Task SearchClientsForTenantExample(TenantId tenantId)
    {
        using var client = CamundaClient.Create();

        var result = await client.SearchClientsForTenantAsync(
            tenantId,
            new SearchClientsForTenantRequest());

        foreach (var c in result.Items)
        {
            Console.WriteLine($"Client: {c.ClientId}");
        }
    }
    // </SearchClientsForTenant>
    #endregion SearchClientsForTenant

    #region SearchGroupIdsForTenant

    // <SearchGroupIdsForTenant>
    public static async Task SearchGroupIdsForTenantExample(TenantId tenantId)
    {
        using var client = CamundaClient.Create();

        var result = await client.SearchGroupIdsForTenantAsync(
            tenantId,
            new TenantGroupSearchQueryRequest());

        foreach (var group in result.Items)
        {
            Console.WriteLine($"Group: {group.GroupId}");
        }
    }
    // </SearchGroupIdsForTenant>
    #endregion SearchGroupIdsForTenant

    #region SearchRolesForTenant

    // <SearchRolesForTenant>
    public static async Task SearchRolesForTenantExample(TenantId tenantId)
    {
        using var client = CamundaClient.Create();

        var result = await client.SearchRolesForTenantAsync(
            tenantId,
            new RoleSearchQueryRequest());

        foreach (var role in result.Items)
        {
            Console.WriteLine($"Role: {role.Name}");
        }
    }
    // </SearchRolesForTenant>
    #endregion SearchRolesForTenant

    #region SearchMappingRulesForTenant

    // <SearchMappingRulesForTenant>
    public static async Task SearchMappingRulesForTenantExample(TenantId tenantId)
    {
        using var client = CamundaClient.Create();

        var result = await client.SearchMappingRulesForTenantAsync(
            tenantId,
            new MappingRuleSearchQueryRequest());

        foreach (var rule in result.Items)
        {
            Console.WriteLine($"Mapping rule: {rule.MappingRuleId}");
        }
    }
    // </SearchMappingRulesForTenant>
    #endregion SearchMappingRulesForTenant
}
